pkg/resources/secret_with_basic_authentication.go

```diff
@@ -107,54 +133,55 @@ func ReadContextSecretWithBasicAuthentication(ctx context.Context, d *schema.Res
 	return nil
 }
 
-func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
 	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	if request := handleSecretUpdate(id, d); request != nil {
-		if err := client.Secrets.Alter(ctx, request); err != nil {
-			return diag.FromErr(err)
-		}
+	set := &sdk.SecretSetRequest{}
+	unset := &sdk.SecretUnsetRequest{}
+	handleSecretUpdate(d, set, unset)
+	setForFlow := &sdk.SetForFlowRequest{
+		SetForBasicAuthentication: &sdk.SetForBasicAuthenticationRequest{},
 	}
 
 	if d.HasChange("username") {
 		username := d.Get("username").(string)
-
-		request := sdk.NewAlterSecretRequest(id)
-		setRequest := sdk.NewSetForBasicAuthenticationRequest().WithUsername(username)
-		request.WithSet(*sdk.NewSecretSetRequest().WithSetForBasicAuthentication(*setRequest))
-
-		if err := client.Secrets.Alter(ctx, request); err != nil {
-			return diag.FromErr(err)
-		}
+		setForFlow.SetForBasicAuthentication.WithUsername(username)
+		set.WithSetForFlow(*setForFlow)
 	}
 
 	if d.HasChange("password") {
 		password := d.Get("password").(string)
+		setForFlow.SetForBasicAuthentication.WithPassword(password)
+		set.WithSetForFlow(*setForFlow)
+	}
 
-		request := sdk.NewAlterSecretRequest(id)
-		setRequest := sdk.NewSetForBasicAuthenticationRequest().WithPassword(password)
-		request.WithSet(*sdk.NewSecretSetRequest().WithSetForBasicAuthentication(*setRequest))
+	if !reflect.DeepEqual(*set, sdk.SecretSetRequest{}) {
+		if err := client.Secrets.Alter(ctx, sdk.NewAlterSecretRequest(id).WithSet(*set)); err != nil {
+			return diag.FromErr(err)
+		}
+	}
 
-		if err := client.Secrets.Alter(ctx, request); err != nil {
+	if !reflect.DeepEqual(*unset, sdk.SecretUnsetRequest{}) {
+		if err := client.Secrets.Alter(ctx, sdk.NewAlterSecretRequest(id).WithUnset(*unset)); err != nil {
 			return diag.FromErr(err)
 		}
 	}
 
 	return ReadContextSecretWithBasicAuthentication(ctx, d, meta)
 }
 
-func DeleteContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func DeleteContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
```

Comment:
All of these Delete functions are the same and can be moved to common.

Reply:
done